food-waste-express-api/routes/home.js

```diff
@@ -134,15 +138,35 @@ router.post('/:objectId', async (req, res) => {
     
     try {
         let userBasket = await getUserBasket(currentUserId) //returns first instance
+        let tempBasket = []
+
         if (userBasket != null) {
-            userBasket.addUnique("basketOfProductId", {productId, quantity})
-            await userBasket.save()
+            let doesNotHaveDuplicate = true;
+            if (userBasket.get("basketOfProducts").length > 0) {
+                for (productDetail of userBasket.get("basketOfProducts")) {
```

Comment:
Used userBasket.get("basketOfProducts" twice, save as its own variable. Maybe name it allProductsInUserBasket?